src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -142,12 +142,18 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
           }
           break;
         case FORMAT_NAME_NODES :
-          if (!liveState.isNameNode1Initialized()) {
+          if (!liveState.isNameNode1Initialized() && !liveState.isNameNode2Initialized()) {
             sendMessageTo(
                 driver,
                 liveState.getFirstNameNodeTaskId(),
                 liveState.getFirstNameNodeSlaveId(),
                 HDFSConstants.NAME_NODE_INIT_MESSAGE);
+          } else if (!liveState.isNameNode1Initialized()) {
+            sendMessageTo(
+                driver,
+                liveState.getFirstNameNodeTaskId(),
+                liveState.getFirstNameNodeSlaveId(),
+                HDFSConstants.NAME_NODE_BOOTSTRAP_MESSAGE);
```

Comment:
Just checking: Is Bootstrap the right message to send here? If I understand correctly, the scenario covered here is if NN1 was init'd and NN2 was bootstrapped, then NN1 died and had to be restarted (either on same/different node). Does it then get bootstrapped instead of initialized?
And what if they were init'd (NN1) and bootstrapped (NN2), and then both died. Do you really want to send INIT to the new NN1 again?